ts/data.ts

```diff
@@ -1,6 +1,75 @@
-const data = {
+interface Data {
+  view: string;
+  entries: Entry[];
+  editing: null;
+  nextEntryId: number;
+}
+
+const data: Data = {
   view: 'entry-form',
   entries: [],
   editing: null,
   nextEntryId: 1,
 };
+
+function writeData(): void {
+  const dataJSON = JSON.stringify(data);
+  localStorage.setItem('data-storage', dataJSON);
+}
+
+function readData(): [] {
+  const dataJSON = localStorage.getItem('data-storage');
+  if (dataJSON) {
+    return JSON.parse(dataJSON);
+  } else {
+    return [];
```

Comment:
This won't be an empty array. You need to create the initial state of your `data` object